testing/integration/endpoint_security_test.go

```diff
@@ -58,6 +61,217 @@ var protectionTests = []struct {
+}
+
+// The steps in this test are the following
+// * Prepare a fixture with an older version of the agent
+// * Create a generic policy
+// * Generate an enrollment token
+// * Add the elastic defend package to the policy
+// * Update the policy to set "is_protected" to true
+// * Get the uninstall token for the policy and store it in the fixture
+// * Install and enroll the fist agent
+// * Validate the health of both the agent and endpoint
+// * Validate that tamper protection is enabled in endpoint by trying to
+// uninstall it
+// * Setup a fixture with the test build
+// * Install the new agent
+// * Validate that the new agent and endpoint are healthy
+// * Validate that the initial endpoint version is smaller than that of the
+// upgraded one
+// * Validate that tamper protection is still enabled by trying to uninstall
+// endpoint
+// * Uninstall endpoint using the uninstall token
+// * Validate that endpoint is indeed uninstalled
+// * Cleanup endpoint
+// * Cleanup agent
+//
+// For cleaning up the agent the test relies on the first installation where the
+// cleanup functions are set
+func testTamperProtectedDebRpmUpgrades(t *testing.T, info *define.Info, packageFormat string) {
+	ctx := context.Background()
+	upgradeFromVersion, err := upgradetest.PreviousMinor()
+	require.NoError(t, err)
+	t.Logf("Preparing fixture with agent version %s", upgradeFromVersion.String())
+
+	startFixture, err := atesting.NewFixture(
+		t,
+		upgradeFromVersion.String(),
+		atesting.WithFetcher(atesting.ArtifactFetcher()),
+		atesting.WithPackageFormat(packageFormat),
+	)
+	require.NoError(t, err)
+	startFixture.Prepare(ctx)
+
+	t.Log("Creating a generic policy and enrollment token")
+	policy := createBasicPolicy()
+	policyResp, enrollKeyResp := createPolicyAndEnrollmentToken(ctx, t, info.KibanaClient, policy)
+
+	t.Log("Install elastic defend")
+	pkgPolicyResp, err := installElasticDefendPackage(t, info, policyResp.ID)
+	require.NoErrorf(t, err, "Policy Response was: %v", pkgPolicyResp)
+
+	isProtected := true
+	updateReq := kibana.AgentPolicyUpdateRequest{
+		Name:        policy.Name,
+		Namespace:   policy.Namespace,
+		IsProtected: &isProtected,
+	}
+
+	t.Log("Updating the policy to set \"is_protected\" to true")
+	_, err = info.KibanaClient.UpdatePolicy(ctx, policyResp.ID, updateReq)
+
+	t.Log("Get the policy uinstall token and store it in the fixture")
+	tools.SetPolicyUninstallTokenInFixture(ctx, t, info.KibanaClient, startFixture, policyResp.ID)
+
+	opts := atesting.InstallOpts{}
+	t.Log("Install and enroll the first agent")
+	tools.InstallAgentForPolicyWithToken(ctx, t, opts, startFixture, info.KibanaClient, policyResp.ID, enrollKeyResp)
+
+	t.Cleanup(func() {
+		_, err := os.Stat("/opt/Elastic/Endpoint/elastic-endpoint")
+		if os.IsNotExist(err) {
+			t.Log("Endpoint binary does not exist, aborting endpoint cleanup")
+			return
+		}
+
+		out, err := exec.Command("sudo", "systemctl", "stop", "ElasticEndpoint").CombinedOutput()
```

Comment:
Again, why not just uninstall agent via the deb/rpm? 

If you aren't uninstall the agent, agent is going to install endpoint again immediately after you do this if endpoint is still in the policy.